auspice/server/getDatasetHelpers.js

```diff
@@ -36,9 +36,26 @@ const splitPrefixIntoParts = (prefix) => {
   /* The first part of the prefix is an optional source name.  The rest is the
    * source path parts.
    */
-  const sourceName = sources.has(prefixParts[0])
-    ? prefixParts.shift()
-    : "core";
+  let sourceName;
+
+  switch (prefixParts[0]) {
+    case "core":
+      sourceName = prefixParts.shift();
+      break;
+    case "community":
+      sourceName = prefixParts.shift();
+      break;
+    case "staging":
+      sourceName = prefixParts.shift();
+      break;
```

Comment:
In case you don't know, when two `case` statements in a `switch` should do the same thing, you can combine them. For example:

```js
case "community":
case "staging":
  sourceName = prefixParts.shift();
  break;
```

This is because `case` statements that don't include a `break` statement will "fall through" to the next `case`.

Reply:
I didn't know that. Thanks!